imap/index.c

```diff
@@ -1928,7 +1927,7 @@ static int needs_modseq(const struct searchargs *searchargs,
 {
     int i;
 
-    if (search_expr_uses_attr(searchargs->root, "modseq"))
+    if (searchargs->state & GETSEARCH_MODSEQ)
```

Comment:
I'm surprised you're not getting a warning here about bitwise-and in a condition being a possible typo?  Usually these want an extra set of parens to convince the compiler that you meant what you wrote, and not `&&`, like this:

```suggestion
    if ((searchargs->state & GETSEARCH_MODSEQ))
```

Though I just tried building this branch and didn't get a warning for it either, so maybe gcc doesn't warn about that anymore, or is now smart enough to not warn when one of the operands is clearly a bitmask.  Curious!

Reply:
I think I've been using constructs like this for a while and I see numerous uses throughout our code, so I think we're good here